introduction_to_amazon_algorithms/object2vec_movie_recommendation/object2vec_movie_recommendation.ipynb

```diff
@@ -144,11 +133,29 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%%bash\n",
+    "# Copy original downloaded data from s3 bucket.\n",
+    "\n",
+    "!mkdir ml-100k\n",
+    "downloaded_data_bucket = \"penny-cache-alpha-us-west-2\"\n",
+    "downloaded_data_prefix = \"1p-notebooks/data/ml-100k\"\n",
+    "\n",
+    "import boto3\n",
+    "import os\n",
     "\n",
-    "curl -o ml-100k.zip http://files.grouplens.org/datasets/movielens/ml-100k.zip\n",
-    "unzip ml-100k.zip\n",
-    "rm ml-100k.zip"
+    "\n",
+    "def download_all_objects_in_folder(bucket, dir_path, output_folder):\n",
+    "    s3_resource = boto3.resource(\"s3\")\n",
+    "    my_bucket = s3_resource.Bucket(bucket)\n",
+    "    objects = my_bucket.objects.filter(Prefix=dir_path + \"/\")\n",
+    "    for obj in objects:\n",
+    "        print(obj.key)\n",
+    "        if obj.key[-1] == \"/\":\n",
```

Comment:
Is `obj.key` the filename? I'm assuming then that the `if` condition is checking if it's a directory then. In this case, I think we can use `if obj.key.endswith("/"):` instead.